experiments/sharedUtils/BoxGenerator.js

```diff
@@ -1,127 +1,157 @@
 // Object to store set of biconditionals
-function AndObj(arg){
+function AndObj(arg, comps){
     this.arg = arg
+    this.comps = comps
     this.type = "and"
     this.eval = function(args){
-        for(let i = 0; i < arg.length; i++){
-            if(arg[i] !== args[i]) return false
+        for(var i = 0; i < comps.length; i++){
+            if(arg[comps[i]] !== args[comps[i]]) return false
         }
         return true
     }
 }
 
 // Object to store set of disjunctions
-function OrObj(arg){
+function OrObj(arg, comps){
     this.arg = arg
+    this.comps = comps
     this.type = "or"
     this.eval = function(args){
-        for(let i = 0; i < arg.length; i++){
-            if(arg[i] === args[i]) return true
+        for(var i = 0; i < comps.length; i++){
+            if(arg[comps[i]] === args[comps[i]]) return true
         }
         return false
     }
 }
 
 // Object to store a single conditional
-function SingObj(arg){
+function SingObj(arg, comps){
     this.arg = arg
+    this.comps = comps
     this.type = "sing"
     this.eval = function(args){
-        for(let i = 0; i < arg.length; i++){
-            if(arg[i] && args[i]) return true
-        }
-        return false
+        return arg[comps[0]] === args[comps[0]]
     }
 
 }
 
 //stores the configurations that will control (turn on) lights
-function Config(arg){
+function Config(arg, numButtons){
     this.controls = arg
-}
-
-// stores the current button settings
-function Setting(arg){ // buttons currently pressed
-    this.args = arg
+    this.numButtons = numButtons
 }
 
 //generate the box. Takes in a config (what turns on what) and returns a function that
 // determine what is turned on
 function generateBox(config){
     //function to return, which will take in a setting and return what is turned on
+    function configDict(){
+        let dict = {} // keys: setting of buttons, values: lights
+        for(var i = 0; i < (Math.pow(2, config.numButtons)); i++){
```

Comment:
`configDict` is a little hard to understand, but as long as you've verified that it produces `2*config.numButtons` unique binary strings for different numbers of buttons, it'll do - maybe some comments to explain what's going on here?